solr/solrj/src/java/org/apache/solr/client/solrj/impl/CloudHttp2SolrClient.java

```diff
@@ -245,6 +249,16 @@ public Builder withParallelUpdates(boolean parallelUpdates) {
       return this;
     }
 
+    /**
+     * This is the time to wait to refetch the state after getting the same state version from ZK
+     *
+     * <p>secs
+     */
+    public Builder setRetryExpiryTime(int secs) {
```

Comment:
No "with" terminology?

Reply:
Can you incorporate TimeUnit as a parameter so that it's clearer what the unit is at the call site?
IntelliJ tip: do this by selecting "TimeUnit.SECONDS" on the line below and doing cmd-opt-P (introduce parameter)

Reply:
are you suggesting the method look like "public Builder setRetryExpiryTime(int secs, TimeUnit seconds)"

Reply:
we do have the "int secs" or "int connectionTimeoutMillis" as parameters on various methods.   I wonder if we should have another JIRA to go through and make all timeunits on builders be the same????

Reply:
@dsmiley I added https://issues.apache.org/jira/browse/SOLR-16590 to track standardizing across all builders the naming format of the methods.

Reply:
TimeUnit class was introduced in part to add clarity to call-sites of a method so the unit is clear.  `blah.setTime(TimeUnit.SECOND, 1)` is fine as well as `blah.setTime(TimeUnit.MINUTE,2)` -- the caller picks the unit convenient to them.  With that design, the method is designed unit-free -- definitely NOT with variables named "second" as you proposed since the unit could be anything.  Internally (implementation of the setter), we need to pick a unit to standardize to on some internal field to store the result, and name the field to be clear as to what the internal unit chosen is. (e.g. retryExpirySecs).  Again, that's *internal*, the caller choses a unit convenient to them.

Reply:
I have added https://issues.apache.org/jira/browse/SOLR-16595 to do this over all builders...